### PR TITLE
New dashboard: notification to point users there

### PR DESCRIPTION
**Pull request by humitos**
We can add the notifications to all the users by running the following code:

```python
from readthedocs.core.notifications import MESSAGE_BETA_DASHBOARD_AVAILABLE

for user in User.objects.all():
    Notification.objects.add(
      message_id=MESSAGE_BETA_DASHBOARD_AVAILABLE,
      attached_to=user,
      dismissable=True,
    )
```

This will show a one-time dismissable notification to these users.

In the future (weeks, months) if we want to show them another notification like this
one as a reminder, we can run the same code again.

Screenshot_2024-02-27_18-42-27

Related https://github.com/readthedocs/ext-theme/pull/183
Related https://github.com/readthedocs/ext-theme/issues/284

### CODE REVIEW

**Review by agjohnson — COMMENTED** [review]
I have questions on the UX here, and ultimately still lean towards adding a link on the build page next.

I worry that this notification will be visible on the new dashboard as users log in to the instance or click the notification to log in. Is there a way that we can filter this out from the new dashboard?

**Comment by humitos** [comment]
> I worry that this notification will be visible on the new dashboard as users log in to the instance or click the notification to log in. Is there a way that we can filter this out from the new dashboard?

I updated the notification to show a different message when it's show on the new dashboard.

### Old dashboard

It says the new dashboard is available with a link to it.

Screenshot_2024-02-28_12-55-05


### New dashboard

It says there may be issues and ask for feedback to the user.

Screenshot_2024-02-28_12-54-30

File: readthedocs/core/notifications.py [hunk 1]
```diff
@@ -22,6 +23,19 @@
...
+        body=_(
+            textwrap.dedent(
+                """
+                We are launching our new <strong>beta dashboard</strong> with a fresh redesign and new powerful features.
+                <a href="https://beta.readthedocs.org/">Give it a try</a> and send us feedback.
+            """
+            ).strip(),
+        ),
+        type=INFO,
+    ),
```
[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocs/core/notifications.py:44]: Yeah, not suggesting that the link clear the notification, just looking for clarification on the UX.

Mutating the message for the dashboard avoids this weird scenario :+1: 

> Just run the code I put in the description from the production Django shell.

I saw this, but the plan seemed rather basic. I think we need to be more careful with notifications if the user dismisses the first notification (don't spam the users).

We should plan a bit more here, and I feel we should not do all users at once.

**Review by agjohnson — APPROVED** [review]

File: readthedocs/core/notifications.py [hunk 2]
```diff
@@ -22,6 +23,25 @@
...
+            textwrap.dedent(
+                """
+                {% if RTD_EXT_THEME_ENABLED %}
+                This dashboard is currently in beta,
+                you can <a href="https://{{ PRODUCTION_DOMAIN }}">return to the legacy dashboard</a> if you encounter any problems.
+                Feel free to <a href="https://{{ PRODUCTION_DOMAIN }}/support/">report any feedback</a> you may have.
+                {% else %}
+                Our new <strong>beta dashboard</strong> is now available for testing.
+                <a href="https://beta.{{ PRODUCTION_DOMAIN }}/">Give it a try</a> and send us feedback.
+                {% endif %}
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/core/notifications.py:39]: > This is fantastic

"Fantastic" is a big word, tho. I think saying a "nice hack" fits better here 😝

**Pull request merged**